syntax = "proto3";

package prot;

service Myserver {
    rpc GetKeys (Keys_Request) returns (Array_Reply) {}
    rpc GenerateJson (Generate_Request) returns (Simple_Json_Reply) {}
    rpc FindValue (Find_Request) returns (Array_Reply) {}
    rpc Convert (Convert_Request) returns (Convert_Reply) {}

    rpc Test(Empty_message) returns (Array_Reply){}
}
message Empty_message {

}

// messages types
message Keys_Request {
    string json = 1;
}
message Generate_Request {
    int32 level = 1;
    int32 numkeys = 2;
}
message Find_Request {
    string value = 1;
    string json = 2;
}
message Convert_Request {
    string audioformat = 1;
    bytes base64 = 2;
}

// replies types
message Simple_Json_Reply {
    string json = 1;
}
message Array_Reply {
    repeated string arr = 1;
}
message Convert_Reply {
    bytes base64 = 1;
}
message Message_Keys_Reply {
    repeated string keys = 1;
}
